src/module/util.ts

```diff
@@ -63,6 +65,8 @@ const isValue = (v: any): boolean => v || v === 0;
 const isFunction = (v: any): boolean => typeof v === "function";
 const isString = (v: any): boolean => typeof v === "string";
 const isNumber = (v: any): boolean => typeof v === "number";
+const isNumberArray = (arr: any): boolean => isArray(arr) && arr.every(v => isNumber(v));
+const isRange = (arr: any): boolean => isNumberArray(arr) && arr.length === 2;
```

Comment:
util.ts is the collection for general purposes. Ranged array data is only used for bar type, so for now including this in a global utility file might not necessary.

Reply:
Okay I'll move the code to a bar related place.